pkg/platform/api/buildplanner/model/buildplan.go

```diff
@@ -65,30 +70,134 @@ func (o Operation) String() string {
 }
 
 // BuildPlan is the top level object returned by the build planner. It contains
-// the project, commit, and build.
-type BuildPlan struct {
+// the commit and build.
+type BuildPlan interface {
+	Build() (*Build, error)
+	CommitID() (strfmt.UUID, error)
+}
+
+func NewBuildPlanResponse(owner, project string) BuildPlan {
+	if owner != "" && project != "" {
+		return &BuildPlanByProject{}
+	}
+	return &BuildPlanByCommit{}
+}
+
+type BuildPlanByProject struct {
 	Project *Project `json:"project"`
+	*Error
+}
+
+func (b *BuildPlanByProject) Build() (*Build, error) {
+	if b.Project == nil {
+		return nil, errs.New("Project is nil")
+	}
+
+	if b.Project.Error != nil && b.Project.Message != "" {
+		return nil, errs.New(b.Project.Message)
+	}
+
+	if b.Project.Commit == nil {
+		return nil, errs.New("Commit is nil")
+	}
+
+	if b.Project.Commit.Error != nil && b.Project.Commit.Message != "" {
+		return nil, errs.New(b.Project.Commit.Message)
+	}
+
+	if b.Project.Commit.Type == NotFound {
+		return nil, locale.NewError("err_buildplanner_commit_not_found", "Build plan does not contain commit")
+	}
+
+	if b.Project.Commit.Build == nil {
+		if b.Project.Commit.Error != nil {
+			return nil, errs.New("Commit not found: %s", b.Project.Commit.Error.Message)
```

Comment:
This doesn't seem like it would ever trigger as line 105 would have already caught this.